### PR TITLE
Update macro to use by default for pallets

### PR DESCRIPTION
**Pull request by NukeManDan**
> Just FYI for all pallet developers and reviewers, all new pallets should have:
> 
> ```rust
> #[pallet::pallet]
> #[pallet::generate_store(pub(super) trait Store)]
> #[pallet::generate_storage_info]            <--- THIS LINE
> pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
> 
> Which will help ensure all of your storage items are bounded
> Old pallets do not have this, and we need to migrate them, but new pallets have no excuse :)
> We should consider flipping the behavior so that it is implicit it rather than needing to include it

From @shawntabrizi :point_up:
I will also update the template & example pallet in substrate & update the templates we own in this org, once this PR is apprved.

### CODE REVIEW

File: v3/docs/03-runtime/b-macros/index.mdx [hunk 1]
```diff
@@ -180,13 +180,39 @@ This is an example of its definition:
...
 #[pallet::pallet]
 #[pallet::generate_store(pub(super) trait Store)]
-pub struct Pallet<T>(PhantomData<T>);
+#[pallet::generate_storage_info]
+pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);
 ```
 
 **Docs**
 
 - See the [macro expansion](/rustdocs/latest/frame_support/attr.pallet.html#macro-expansion-1) added to the `struct Pallet<T>` definition
```
Comment (NukeManDan) [inline v3/docs/03-runtime/b-macros/index.mdx:190]: Below is new content to review

**Comment by sacha-l** [comment]
@NukeManDan : AFAIK the `pub struct Pallet<T, I = ()>(PhantomData<(T, I)>);` syntax is to handle pallet instantiability - not sure why we'd want that in all tutorials ? I think what's needed is just to add the `#[pallet::generate_storage_info]` line only.

**Comment by NukeManDan** [comment]
Ah - makes sense to me. Should this instantiable syntax be highlighted somewhere in the docs I am not seeing? I can update to no PhantomData.

**Review by sacha-l — CHANGES_REQUESTED** [review]
I think instantiable pallets fall outside the scope of what this PR is trying to achieve, which is adding `#[pallet::generate_storage_info]`. On a separate note, would be nice to update [this guide](https://docs.substrate.io/how-to-guides/v3/basics/instantiable-pallets/) at some point!

**Review by sacha-l — APPROVED** [review]

**Pull request merged**